python/osrd_schemas/scripts/truncate_infra.py

```diff
@@ -0,0 +1,378 @@
+            ]
+
+            res.append(route)
+
+        elif not entry_included and not exit_included and n_switches_included > 0:
+            # Some edge case that should be included but isn't handled yet
+            print(f"Weird route {route.id} that should be included but isn't handled")
+    return res
+
+
+def order_switch_directions(
+    route: Route,
+    detectors_by_id: dict[str, Detector],
+    buffer_stops_by_id: dict[str, BufferStop],
+    switches_by_id: dict[str, Switch],
+) -> list[tuple[str, str]]:
+    """Order the switches and their directions in the route from the entry point to the exit point."""
+    res = []
+
+    tracks_by_switch = {
+        switch_id: [port.track for port in switch.ports.values()]
+        for switch_id, switch in switches_by_id.items()
+    }
+
+    switch_directions = list(route.switches_directions.items())
+    entry_waypoint = (
+        detectors_by_id[route.entry_point.id]
+        if route.entry_point.type == "Detector"
+        else buffer_stops_by_id[route.entry_point.id]
+    )
+    track_id = entry_waypoint.track
+    while switch_directions != []:
+        switch_dir_on_track = [
+            (switch_id, dir)
+            for switch_id, dir in switch_directions
+            if track_id in tracks_by_switch[switch_id]
+        ]
+
+        assert len(switch_dir_on_track) == 1
+        res.append(switch_dir_on_track[0])
+        switch_id, dir = res[-1]
+
+        switch_directions.remove(res[-1])
+
+        switch = switches_by_id[switch_id]
+        track_id = get_connected_track_id(switch, track_id, dir)
+    return res
+
+
+def get_connected_track_id(switch: Switch, current_track_id: str, group: str) -> str:
+    """Get the id of the track connected to the given track by this switch."""
+    switch_type = SWITCH_TYPES[switch.switch_type]
+    connections = switch_type.groups[group]
+
+    connected_track = None
+    for connection in connections:
+        if switch.ports[connection.src].track == current_track_id:
+            connected_track = switch.ports[connection.dst].track
+            break
+        elif switch.ports[connection.dst].track == current_track_id:
+            connected_track = switch.ports[connection.src].track
+            break
+    if connected_track is None:
+        raise ValueError(f"Could not find the connected track for {current_track_id}")
+    return connected_track
+
+
+if __name__ == "__main__":
+    import json
+    from argparse import ArgumentParser, RawTextHelpFormatter
+    from pathlib import Path
+
+    parser = ArgumentParser(formatter_class=RawTextHelpFormatter)
+    parser.add_argument("input", help="Input RailJson file", type=Path)
+    parser.add_argument("output", help="Output RailJson file", type=Path)
+    parser.add_argument(
+        "bbox",
+        help="Bounding box as GeoJSON polygon coordinates\n"
+        'For example "[[0, 0], [0, 1], [1, 1], [1, 0]]"\n'
+        "A way to get it is to use https://geojson.io/",
```

Comment:
Maybe a bit nitpicky, but a "real" bounding box is only made of 2 coords, for the same example it would be `[[0,0],[1,1]]`.
As the feature is finer, I'd change the example at least, maybe even rename the param:
```suggestion
        "bounds",
        help="Boundaries as GeoJSON polygon coordinates\n"
        'For example ♦️: "[[0, 0], [1, 1], [2, 0], [1, -1]]"\n'
        "A way to get it is to use https://geojson.io/",
```

Reply:
I’ve slightly diverged from your comment since I used “boundary” instead of “bound” but I’ve made the change in https://github.com/OpenRailAssociation/osrd/pull/13526/commits/ff205ec667afbad1f16fee15e325cc5196e87e43.